### PR TITLE
Fix strawman compilation v2

### PR DESCRIPTION
**Pull request by odersky**
Some more fixes related to making the latest version of the strawman (and some random variations) compile under dotty.

### CODE REVIEW

**Comment by odersky** [comment]
I give up. I tried rebase + git submodule several times. No avail. Why are we doing this?

**Comment by odersky** [comment]
@nicolasstucki @smarter Can one of you please take over this branch and make it pass the CI? Thanks!

**Comment by odersky** [comment]
Here's some more info:

/Users/odersky/workspace/dotty> git submodule update --init
fatal: No url found for submodule path 'scala-scala' in .gitmodules
/Users/odersky/workspace/dotty> cat .git
.git/           .gitattributes  .gitignore      .gitmodules     
/Users/odersky/workspace/dotty> cat .gitmodules 
[submodule "scala-backend"]
	path = scala-backend
	url = https://github.com/lampepfl/scala.git
	branch = sharing-backend
[submodule "scala2-library"]
	path = scala2-library
	url = https://github.com/lampepfl/scala.git
	branch = dotty-library
[submodule "collection-strawman"]
	path = collection-strawman
	url = https://github.com/scala/collection-strawman.git
	branch = master
/Users/odersky/workspace/dotty>

**Comment by smarter** [comment]
I've removed the `scala-scala` submodule that was apparently added in this PR.

File: docs/docs/internals/syntax-summary.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
+---
+title: Syntax Summary
+layout: default
+chapter: 13
+---
+
+# Syntax Summary
```
Comment (smarter) [inline docs/docs/internals/syntax-summary.md]: This seems to be copy-paste from scalac, and to duplicate the existing `syntax.md`. Did you intend to add this file?

**Comment by smarter** [comment]
PR is now green :).

**Comment by odersky** [comment]
Thanks for the fix, @smarter! The spurious scala-scala submodule started showing up in my git status, I believe it was after scala-library was added. I thought I needed to push it like the others. No idea why it showed up suddenly - I did not touch the directory for a while.

File: docs/docs/internals/syntax-summary.md (hunk 1 above)
Comment (odersky) [inline docs/docs/internals/syntax-summary.md]: No, that was a mistake. I'll remove it.

**Comment by smarter** [comment]
@odersky If you still have a `scala-scala` directory around, you should remove it, it's been replaced by the `scala2-library` submodule.

**Comment by odersky** [comment]
@smarter Good to know!

**Pull request merged**